src/bentoml_cli/models.py

```diff
@@ -162,6 +167,20 @@ def delete(delete_targets: str, yes: bool) -> None:  # type: ignore (not accesse
             * Bulk delete multiple models by name and version, separated by " ", e.g.: `bentoml models delete iris_clf:v1 iris_clf:v2`
             * Bulk delete without confirmation, e.g.: `bentoml models delete IrisClassifier --yes`
         """  # noqa
+        from bentoml.exceptions import BentoMLException
+
+        def check_model_is_used(tag: Tag) -> None:
+            in_use: list[Tag] = []
+            for bento in bento_store.list():
+                if bento._model_store is not None:
+                    continue
+                if any(model.tag == tag for model in bento.info.models):
+                    in_use.append(bento.tag)
+            if in_use:
+                raise BentoMLException(
+                    f"Model {tag} is being used by the following bentos and can't be deleted:\n  "
+                    + "\n  ".join(map(str, in_use))
+                )
```

Comment:
Nope, it's required due to this change, to prevent a model from being deleted when it's still used by a bento, as it is no longer stored with the bento(changed by this PR).

Reply:
got it make sense